src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -796,6 +853,33 @@ void SettingsLoader::_addUserProfileParent(const winrt::com_ptr<implementation::
     }
 }
 
+void SettingsLoader::_addOrMergeUserColorScheme(const winrt::com_ptr<implementation::ColorScheme>& newScheme)
+{
+    if (const auto [it, inserted] = userSettings.colorSchemes.emplace(newScheme->Name(), newScheme); !inserted)
+    {
+        // This scheme was not inserted because one already existed.
+        auto existingScheme{ it->second };
+        if (existingScheme->Origin() == OriginTag::User) // we only want to impose ordering on User schemes
+        {
+            it->second = newScheme; // Stomp the user's existing scheme with the one we just got (to make sure the right Origin is set)
+            if (!existingScheme->IsEquivalentForSettingsMergePurposes(newScheme))
+            {
+                hstring newName{ fmt::format(L"{} (modified)", existingScheme->Name()) };
+                int differentiator = 2;
+                while (userSettings.colorSchemes.contains(newName))
+                {
+                    newName = hstring{ fmt::format(L"{} (modified {})", existingScheme->Name(), differentiator++) };
+                }

```

Comment:
Shouldn't this result in `fixup == true`?

Reply:
(This is my only question before approval btw. The others ones are nits.)

Reply:
Great catch! I've added a test for this and yes, it fails :)